src/main/groovy/com/google/protobuf/gradle/GenerateProtoTask.groovy

```diff
@@ -203,10 +206,37 @@ public abstract class GenerateProtoTask extends DefaultTask {
   }
 
   static int getCmdLengthLimit(String os) {
-    if (os != null && os.toLowerCase(Locale.ROOT).indexOf("win") > -1) {
-      return WINDOWS_CMD_LENGTH_LIMIT
+    return isWindows(os) ? WINDOWS_CMD_LENGTH_LIMIT : Integer.MAX_VALUE
+  }
+
+  static boolean isWindows(String os) {
```

Comment:
These methods should be `private` but CodeNarc has some conflicting rules if they are `private`, so they are `package private`. I'm not overly familiar with groovy so please advise if there is a better way to go about this.

put these methods at the bottom and get:
> StaticMethodsBeforeInstanceMethods	Enforce that all static methods are above all instance methods within a class

put these methods after other package private static methods and get:
>  PublicMethodsBeforeNonPublicMethods	Enforce that all public methods are above protected and private methods.